addons/web/static/src/core/utils/files.js

```diff
@@ -26,3 +27,30 @@ export function checkFileSize(fileSize, notificationService) {
     }
     return true;
 }
+
+/**
+ * Hook to upload a file to the server.
+ * @returns {function}
+ */
+export function useFileUploader() {
+    const http = useService("http");
```

Comment:
Can I use "useService" inside hooks? I thought not as "checkFileSize" is requesting the "notificationService" as a method argument, but maybe it works 🤷‍♂️

Reply:
Yes `useSerice()` inside hooks is good. As long as there's no dependency cycle and the hook is called in `Component.setup()`, you can have hooks call other hooks